app/models/create_expenses_with_receipt.rb

```diff
@@ -0,0 +1,31 @@
+class CreateExpensesWithReceipt
+  attr_reader :expense
+
+  def initialize(expense_params)
+    @params = expense_params
+  end
+
+  def create
+    return false unless picture_exists?
+
+    Expense.transaction do
+      create_expense
+      create_receipt
+    end
+    @expense.persisted?
+  end
```

Comment:
It's being raised by line 25 (`@expense = Expense.create!(@params.except(:picture))`) in the `create_expense` call inside the transaction block.